runtime/src/main/java/org/corfudb/runtime/view/ObjectsView.java

```diff
@@ -157,6 +157,17 @@ public long TXEnd() throws TransactionAbortedException {
             long commitAddress = TransactionalContext.getCurrentContext().commitTransaction();
             MicroMeterUtils.time(Duration.ofMillis(System.currentTimeMillis() - context.getStartTime()),
                     "transaction.duration");
+            if (commitAddress == Address.NON_ADDRESS) {
+                // If no streams were touched or only empty streams were touched, the returned address would be -1
+                // But -1 can be detrimental to stream subscription which is just looking for a safe spot
+                // to resume subscription from, so instead return the address from the snapshot token taken.
+                long snapshotAddress = context.getSnapshotTimestamp().getSequence();
+                if (snapshotAddress > 0) {
+                    // Perform a read at this address to force materialization.
+                    runtime.getAddressSpaceView().peek(snapshotAddress);
```

Comment:
`AddressSpaceView::peek` is not a read, it is a peek. The difference between a peek and a read, is that the latter drives a state machine read, which either hole fills or completes a partial write, but peek doesn't this.

This additional logic has no side-effect, so I'm not sure how the test is passing. The returned value is not even being validated.

Reply:
Did not realize peek skipped the hole fill logic. thanks.